### ISSUE TITLE
ReflectionPad should accept 0-dim batch size.

### ISSUE THREAD

**Issue by v0dro** [pytorch/pytorch#39225]
All flavours of ReflectionPad should accept 0-dim batch size.

Tests:

``` python
def test_ReflectionPad1d_Forward():
    z = torch.ones(0,3,10)
    m = torch.nn.ReflectionPad1d(2)
    m(z)

def test_ReflectionPad1d_Backward():
    z = torch.ones(0,3,10)
    m = torch.nn.ReflectionPad1d(2)
    m(z).sum().backward()

def test_ReflectionPad2d_Forward():
    z = torch.ones(0,3,10,10)
    m = torch.nn.ReflectionPad2d(2)
    m(z)

def test_ReflectionPad2d_Backward():
    z = torch.ones(0,3,10,10)
    m = torch.nn.ReflectionPad2d(2)
    m(z).sum().backward()
```

**Comment by rgommers** [comment]
This is duplicate with `gh-12013` (avoiding cross-linking to keep that issue as clean as possible), I'll close it.
